RULES/cc/toolchain.go

```diff
@@ -105,7 +143,17 @@ func joinQuoted(paths []core.Path) string {
 	return b.String()
 }
 
-var DefaultToolchain = GccToolchain{
+var toolchains = make(map[string]Toolchain)
+
+func (gcc GccToolchain) Register() Toolchain {
```

Comment:
There should be a global `RegisterToolchain` function. Otherwise there will be no way to register a new (not necessarily GCC) toolchain, since the `toolchains` map is private (which is good). The function should also check for conflicting names. This function can still exist for convenience (and should then call the global register function).

Reply:
Yeah good point